VOKMockUrlProtocol.m

```diff
@@ -111,6 +118,28 @@ - (NSArray *)resourceNames
         [resourceName appendFormat:queryFormat, self.request.URL.query];
     }
     
+    if (headersToEncode) {
+        NSDictionary *headerDict = self.request.allHTTPHeaderFields;
+        NSMutableArray <NSString *>*allIncludedHeaders = [NSMutableArray array];
+        for (NSString *headerName in headersToEncode) {
+            NSString *headerValue = headerDict[headerName];
+            if (headerValue) {
+                NSString *headerString = [NSString stringWithFormat:@"%@=%@", headerName, headerValue];
+                [allIncludedHeaders addObject:headerString];
+            }
+        }
```

Comment:
Instead of building up a string, how about building up a dictionary mapping header names to values, then bencoding it to put into the resource name?

Maybe that'd also allow taking the all-headers dict and filtering it rather than iterating over it?